install/Knative-with-ICP.md

```diff
@@ -0,0 +1,173 @@
+the newly created Knative cluster on [IBM Cloud Private](https://www.ibm.com/cloud/private).
+
+You can find [guides for other platforms here](README.md).
+
+## Before you begin
+
+Knative requires a v3.1.1 standard [IBM Cloud Private](https://www.ibm.com/cloud/private) cluster. Before you can install Knative, you must first complete all the steps that are provided in the [IBM Cloud Private standard cluster installation instructions](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html). For example:
+
+1. Install Docker for your boot node only
+
+2. Set up the installation environment
+
+3. Customize your cluster
+
+4. Set up Docker for your cluster nodes
+
+5. Deploy the environment
+
+6. Verify the status of your installation
+
+## Installing Istio
+
+[Follow the instructions to install and run Istio in IBM Cloud Private](https://istio.io/docs/setup/kubernetes/quick-start-ibm/#ibm-cloud-private).
+
+## Installing Knative components
+
+You can install the Knative Serving and Build components together, or individually.
+
+### Installing Knative Serving and Build components
+
+Run the following command to deploy [Knative Serving](https://github.com/knative/serving) and [Knative Build](https://github.com/knative/build)
+
+```shell
+curl -L https://github.com/knative/serving/releases/download/v0.2.2/release-lite.yaml \
+  | sed 's/LoadBalancer/NodePort/' \
+  | kubectl apply --filename -
+```
+
+### Installing Knative Serving only
+
+Replace `release-lite.yaml` to `serving.yaml` file, and then run the following command:
+
+```shell
+curl -L https://github.com/knative/serving/releases/download/v0.2.2/serving.yaml \
+  | sed 's/LoadBalancer/NodePort/' \
+  | kubectl apply --filename -
+```
+
+### Installing Knative Build only
+
+Replace `release-lite.yaml` to `build.yaml` file, and then run the following command:
+
+```shell
+curl -L https://github.com/knative/serving/releases/download/v0.2.2/build.yaml \
+  | sed 's/LoadBalancer/NodePort/' \
+  | kubectl apply --filename -
+```
+
+### Update the image security policy
+You need to update the [image security policy](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/manage_images/image_security.html) to allow the cluster to pull the Knative image when you enable the `image-security-enforcement` at the time of [IBM Cloud Private](https://www.ibm.com/cloud/private) installation.
+
+1. Edit the image security policy.
+    ```
+    kubectl edit clusterimagepolicies ibmcloud-default-cluster-image-policy
+    ```
+
+2. Update `spec.repositories` by adding `"gcr.io/knative-releases/*"`
+    ```yaml
+    spec:
+      repositories:
+      - name: "gcr.io/knative-releases/*"
+    ```
+
+### Update pod security policy
+Configure the namespaces `knative-serving`, `knative-build`, `knative-monitoring` and `knative-eventing` into pod security policy `ibm-privileged-psp`. The step as follows:
```

Comment:
We have only provided steps to install Serving and Build, not Eventing nor Monitoring. This will be confusing to someone new and they might think they actually get those features too.

They should be removed or otherwise, it should be clarified that those policies are optional for if you choose to install them at a later time.

That said, are there issues/vulnerabilities with setting unnecessary security policies?

Reply:
The yaml `release-lite.yaml` include the namespace `knative-monitoring`, so I keep `knative-monitoring` and remove the namespace `knative-eventing` . And I put this section before install  knative.

The security policies here is necessary.  Knative pod can't run normally if the we don't set the security policies.